core/runtime/impl/runtime_external_interface.cpp

```diff
@@ -50,10 +50,9 @@ namespace kagome::runtime {
    */
 
   RuntimeExternalInterface::RuntimeExternalInterface(
-      std::shared_ptr<extensions::Extension> extension,
-      std::shared_ptr<WasmMemory> memory, common::Logger logger)
+      std::shared_ptr<extensions::Extension> extension, common::Logger logger)
       : extension_(std::move(extension)),
-        memory_(std::move(memory)),
+        memory_(extension_->memory()),
```

Comment:
you could save non-const reference to a Memory

Reply:
it is more convenient to work with shared_ptrs. Any strong reasons to switch to references?